Yoast/YoastSEO.js
# Refactor conflicting links assessment

## Summary

This PR can be summarized in the following changelog entry:

* Applies morphological researcher to the text conflicting links assessment.

## Relevant technical choices:

* 

## Test instructions

This PR can be tested by following these steps:

* For English Premium
  * Take a text with a link that does not point at itself and make its text an exact match of your keyphrase. The assessment should return a bad score.
  * Take a text with a link that does not point at itself and make its text a form of your keyphrase or a synonym. The assessment should return a bad score.
  * Check if only full matches trigger a red bullet (i.e., all content words from the link text should be in the keyphrase or a synonym and vice versa)
* For English Free and for all other languages with function-word support
  * Check if function words get trimmed when the assessment is applied and that the word order does not matter.
* For languages without function-word support (Free and Premium)
  * Check if word order does not matter.
  * For Premium: check if synonyms are taken into account.

Note that highlights for this assessment don't work at the moment; this is not a regression: https://github.com/Yoast/YoastSEO.js/issues/1350

Fixes #1825
